src/main/java/seedu/exercise/model/ModelManager.java

```diff
@@ -22,29 +22,34 @@
 
     private final ExerciseBook exerciseBook;
     private final RegimeBook regimeBook;
+    private final ExerciseBook allExerciseBook;
     private final UserPrefs userPrefs;
     private final FilteredList<Exercise> filteredExercises;
+    private final FilteredList<Exercise> suggestedExercises;
     private final FilteredList<Regime> filteredRegimes;
 
     /**
      * Initializes a ModelManager with the given exerciseBook and userPrefs.
      */
-    public ModelManager(ReadOnlyExerciseBook exerciseBook, ReadOnlyRegimeBook regimeBook, ReadOnlyUserPrefs userPrefs) {
+    public ModelManager(ReadOnlyExerciseBook exerciseBook, ReadOnlyRegimeBook regimeBook,
+                        ReadOnlyExerciseBook allExercises, ReadOnlyUserPrefs userPrefs) {
         super();
         requireAllNonNull(exerciseBook, userPrefs);
 
         logger.fine("Initializing with exercise book: " + exerciseBook + " and user prefs " + userPrefs);
 
         this.exerciseBook = new ExerciseBook(exerciseBook);
+        this.allExerciseBook = new ExerciseBook(allExercises);
```

Comment:
I think it will be better if we change the name from ``allExerciseBook`` to ``databaseBook`` as it may confuse others when they see ``exerciseBook`` and ``allExerciseBook`` next to each other.

Reply:
Ok